### DISCUSSION TITLE
Alternative to React Server Components

### DISCUSSION THREAD

**Discussion by wakaztahir** [solidjs/solid-start#1119] [Ideas]
You can already ensure the code runs on server in SolidStart

Here's some benefits of Server Components (the syntax)

PeaceOfMind : Server Components ensure they'll never run on client
Features : Being able to make fetch requests before rendering since its async
Security : Server Components allow you to keep sensitive data and logic on the server, such as tokens and API keys, without the risk of exposing them to the client.
Syntax : Server components have nicer syntax

**Comment by ryansolid** [comment]
To be fair we have had a version of Server Components under a flag(`"islandsRouter": true`) on SolidStart for over a year now but we were not happy where the API was. While this will not be ready to release with 1.0 of SolidStart expect the ability to adopt something equivalent in the future as our Router patterns are already set up for them and we anticipate the move to them to done gradually.
